sdk/maps/Azure.Maps.Search/src/MapsSearchClient.cs

```diff
@@ -456,24 +222,16 @@ public virtual async Task<Response<SearchAddressResult>> SearchAddressAsync(stri
             }
         }
 
-        /// <summary>
-        /// Address geocoding. The geocoding is performed by hitting the geocode endpoint with just the address or partial address in question. The geocoding search index will be queried for everything above the street level data. No POIs will be returned. Note that the geocoder is very tolerant of typos and incomplete addresses. It will also handle everything from exact street addresses or street or intersections as well as higher level geographies such as city centers, counties, states etc.
-        /// </summary>
-        /// <param name="query"> The POI name to search for (e.g., &quot;statue of liberty&quot;, &quot;starbucks&quot;, &quot;pizza&quot;). Must be properly URL encoded. </param>
-        /// <param name="options"> additional options  </param>
+        /// <param name="geocodingBatchRequestBody"> The list of address geocoding queries/requests to process. The list can contain a max of 100 queries and must contain at least 1 query. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual Response<SearchAddressResult> SearchAddress(string query, SearchAddressOptions options = null, CancellationToken cancellationToken = default)
+        /// <exception cref="ArgumentNullException"> <paramref name="geocodingBatchRequestBody"/> is null. </exception>
+        public virtual Response<GeocodingBatchResponse> GetGeocodingBatch(GeocodingBatchRequestBody geocodingBatchRequestBody, CancellationToken cancellationToken = default)
```

Comment:
`GeocodingBatchRequestBody` and `GetGeocodingOptions` only differs in `query` and `optionalId`. I suggest to hide `GeocodingBatchRequestBody` and replace it as following:
* Rename `GetGeocodingOptions` to `GeocodingQuery` and include `string Query` and `string OptionalId` as a member variable
* Mention `OptionalId` only used in batch geocoding in the docstring
* Hide `GeocodingBatchRequestBody`
* In `GetGeocodingBatch`, use `IEnumerable<GeocodingQuery> queries` as argument and convert them into `GeocodingBatchRequestBody`
* Same for reverse geocoding batch request body